From 85e7203d33d02d5fd0bfcacafd53f595773487ba Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Sat, 25 Jan 2025 03:03:03 +0800
Subject: [PATCH 142/258] AOSCOS: MIPS: loongson64: deselect
 ARCH_SUPPORTS_HUGETLBFS

Disable HugeTLB support as it causes TLB exceptions and illegal memory
access on 3A4000 and 3B4000 (possibly more, but we haven't tested other
platforms yet).

Suggested-by: Mingcong Bai <jeffbai@aosc.io>
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 6b38e6da99fd..aae3b2d2b9fd 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -25,7 +25,7 @@ config MIPS
 	select ARCH_USE_MEMTEST
 	select ARCH_USE_QUEUED_RWLOCKS
 	select ARCH_USE_QUEUED_SPINLOCKS
-	select ARCH_SUPPORTS_HUGETLBFS if CPU_SUPPORTS_HUGEPAGES
+	select ARCH_SUPPORTS_HUGETLBFS if CPU_SUPPORTS_HUGEPAGES && !CPU_LOONGSON64
 	select ARCH_WANT_DEFAULT_TOPDOWN_MMAP_LAYOUT if MMU
 	select ARCH_WANT_IPC_PARSE_VERSION
 	select ARCH_WANT_LD_ORPHAN_WARN
-- 
2.48.1

